OfficeDevPnP.Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectPages.cs

```diff
@@ -69,6 +69,12 @@ public override void ProvisionObjects(Web web, ProvisioningTemplate template)
                     web.AddWikiPageByUrl(url);
                     web.AddLayoutToWikiPage(page.Layout, url);
                 }
+
+                if (page.WelcomePage)
+                {
+                   web.SetHomePageByServerRelativePath(url);
```

Comment:
Probably you are right :) The logic for setting the WelcomePage in the ObjectHandler or in the BrandingExtensions is the same. My thoughts has been that being in the BrandingExtension makes it reusable, otherwise it lives only in the ObjectHandler waiting for copy/paste driven development. Last but not least I cannot be sure someone gonna need it as extension. Please advise how to proceed :)